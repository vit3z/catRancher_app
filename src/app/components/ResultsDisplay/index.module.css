.resultsDisplayPaper {
  height: 100%;
  padding: 25px;
  border-radius: 25px;
}

.resultsDisplayGridItem {
  border: 1px solid black;
  height: 137px;
  display: flex;
}