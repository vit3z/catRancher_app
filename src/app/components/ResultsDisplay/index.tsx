import { catImageHeightCalculatorMainDisplay } from "@/app/utils/CatImageHeightCalculatorMainDisplay";
import { catImageWidthCalculatorMainDisplay } from "@/app/utils/CatImageWidthCalculatorMainDisplay";
import { ResultsDisplayProps } from "@/consts/interfaces";
import { Grid, Paper, Typography } from "@mui/material";
import Image from "next/image";
import styles from "./index.module.css";

const ResultsDisplay: React.FC<ResultsDisplayProps> = ({
  foundClowders,
  invalidClowder,
  invalidClowderCause,
}) => {
  return (
    <Paper className={styles.resultsDisplayPaper}>
      <Typography variant="h6">Clowders found</Typography>
      <Grid container>
        {[...Array(12)].map((_, index) => (
          <Grid
            key={index}
            item
            xs={4}
            className={styles.resultsDisplayGridItem}
            alignItems="flex-end"
            justifyContent="center"
          >
            {foundClowders[index] ? (
              <Image
                src={`https://static.quantcast.com/catrancher/${foundClowders[index]}.png`}
                height={
                  catImageHeightCalculatorMainDisplay(foundClowders[index]) / 2
                }
                width={
                  catImageWidthCalculatorMainDisplay(foundClowders[index]) / 2
                }
                alt={`Image of Cat with ID: ${foundClowders[index]}`}
              />
            ) : (
              <></>
            )}
          </Grid>
        ))}
      </Grid>
      {invalidClowder && (
        <div>
          <Typography variant="subtitle1" marginTop="15px" color={"red"} lineHeight={"1.5"}>
            {
              "The cats will NOT get along! Make sure the following properties are not in a 2:1 ratio:"
            }
          </Typography>
          {invalidClowderCause.map((cause: string, index: number) => {
            return (
              <Typography
                key={`disorder-key-${index}`}
                variant="subtitle1"
                color={"red"}
              >
                - {cause.toUpperCase()}
              </Typography>
            );
          })}
        </div>
      )}
    </Paper>
  );
};

export default ResultsDisplay;
