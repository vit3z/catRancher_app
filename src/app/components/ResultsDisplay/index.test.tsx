import React from 'react';
import renderer from 'react-test-renderer';
import ResultsDisplay from './index';
jest.mock('next/image', () => 'img');

describe('ResultsDisplay', () => {
  it('renders correctly when we have a list of valid cats in a clowder', () => {
    const props = {
      foundClowders: ['1ttg', '2wsb', '3brr'],
      invalidClowder: false,
      invalidClowderCause: [],
    };
    const component = renderer.create(<ResultsDisplay {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when a clowder is invalid', () => {
    const props = {
      foundClowders: [],
      invalidClowder: true,
      invalidClowderCause: ['eyes', 'shape'],
    };
    const component = renderer.create(<ResultsDisplay {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with empty clowder', () => {
    const props = {
      foundClowders: [],
      invalidClowder: false,
      invalidClowderCause: [],
    };
    const component = renderer.create(<ResultsDisplay {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
