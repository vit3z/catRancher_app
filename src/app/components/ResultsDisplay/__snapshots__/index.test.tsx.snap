// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsDisplay renders correctly when a clowder is invalid 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 resultsDisplayPaper css-1ps6pg7-MuiPaper-root"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  >
    Clowders found
  </h6>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
  </div>
  <div>
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-2vofx8-MuiTypography-root"
    >
      The cats will NOT get along! Make sure the following properties are not in a 2:1 ratio:
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-bmwhns-MuiTypography-root"
    >
      - 
      EYES
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-bmwhns-MuiTypography-root"
    >
      - 
      SHAPE
    </h6>
  </div>
</div>
`;

exports[`ResultsDisplay renders correctly when we have a list of valid cats in a clowder 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 resultsDisplayPaper css-1ps6pg7-MuiPaper-root"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  >
    Clowders found
  </h6>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    >
      <img
        alt="Image of Cat with ID: 1ttg"
        height={137.5}
        src="https://static.quantcast.com/catrancher/1ttg.png"
        width={90}
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    >
      <img
        alt="Image of Cat with ID: 2wsb"
        height={100}
        src="https://static.quantcast.com/catrancher/2wsb.png"
        width={60}
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    >
      <img
        alt="Image of Cat with ID: 3brr"
        height={90}
        src="https://static.quantcast.com/catrancher/3brr.png"
        width={90}
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
  </div>
</div>
`;

exports[`ResultsDisplay renders correctly with empty clowder 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 resultsDisplayPaper css-1ps6pg7-MuiPaper-root"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  >
    Clowders found
  </h6>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 resultsDisplayGridItem css-r0vdtd-MuiGrid-root"
    />
  </div>
</div>
`;
